flake.nix

```diff
@@ -0,0 +1,41 @@
+{
+  # Override nixpkgs to use the latest set of node packages
+  inputs.nixpkgs.url = "github:NixOS/nixpkgs/master";
+  inputs.systems.url = "github:nix-systems/default";
+
+  outputs =
+    {
+      self,
+      nixpkgs,
+      flake-utils,
+      systems,
+    }:
+    flake-utils.lib.eachSystem (import systems) (
+      system:
+      let
+        pkgs = import nixpkgs { inherit system; };
+      in
+      {
+        devShells.default = pkgs.mkShell {
+          buildInputs = with pkgs; [
+            # generic
+            gnumake
+            gnutar
+
+            # frontend
+            nodejs
+            nodePackages.pnpm
+            nodePackages.typescript
+            nodePackages.typescript-language-server
+
+            # backend
+            go
+            gofumpt
+            golangci-lint
+            go-mockery
+            protobuf
```

Comment:
go install works but that breaks out of the nix environment ... so if the tools are packaged already in the nixpkgs they should be used.

else we have to relay on go install

so that's why there is miss-match in both lists.

Reply:
you can use https://search.nixos.org/packages for lookups online